### PR TITLE
fix(editor): When cloud users click on "How to update your n8n version"  auto-login them before redirecting to the dashboard (no-changelog)

### PR DESCRIPTION
**Pull request by RicardoE105**
## Summary

Title self explanatory.

PR also migrates upgradePanel component to composition API.

image

## Related Linear tickets, Github issues, and Community forum posts

https://linear.app/n8n/issue/ADO-2616/bug-when-cloud-users-click-on-update-info-url-they-dont-get-auto

## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/editor-ui/src/stores/ui.store.ts [hunk 1]
```diff
@@ -749,33 +765,43 @@ export const listenForModalChanges = (opts: {
...
 
 	if (source) {
-		searchParams.set('source', source);
+		url.searchParams.set('source', source);
 	}
+
+	return url.toString();
+};
+
+const generateCloudDashboardAutoLoginLink = async (data: {
```
Comment (mutdmour) [inline packages/editor-ui/src/stores/ui.store.ts:793]: nice refactor. Could we also refactor `redirectToDashboard` in cloud plans store to use this?

File: packages/editor-ui/src/components/UpdatesPanel.vue [hunk 2]
```diff
@@ -1,39 +1,28 @@
...
+const i18n = useI18n();
+
+const nextVersions = computed(() => versionsStore.nextVersions);
+
+const currentVersion = computed(() => versionsStore.currentVersion);
+
+const infoUrl = computed(() => versionsStore.infoUrl);
+
+const onInfoUrlClick = async () => {
+	uiStore.goToVersions();
```
Comment (mutdmour) [inline packages/editor-ui/src/components/UpdatesPanel.vue:24]: lint issue here

**Review by netroy — CHANGES_REQUESTED** [review]
Can we please make the composition API migration a separate PR 🙏🏽?
That way it's much easier to understand what the actual fix is in this PR.

**Comment by RicardoE105** [comment]
Dropping in favor of https://github.com/n8n-io/n8n/pull/11467

**Pull request closed without merging**